R/waapwlspetpeese.R

```diff
@@ -0,0 +1,610 @@
+    jaspResults[["estimates"]] <- estimates
+  } else {
+    estimates <- jaspResults[["estimates"]]
+  }
+
+  # mean estimates
+  if (is.null(estimates[["mean"]]) && options[["estimatesMean"]]) {
+    estimatesMean <- createJaspTable(title = gettextf(
+      "Mean Estimates(%s)",
+      if (options[["measures"]] == "correlation") "\u03C1" else "\u03BC"
+    ))
+    estimatesMean$position <- 1
+    estimates$dependOn(c("estimatesMean"))
+    estimates[["estimatesMean"]] <- estimatesMean
+    estimatesMean <- .wwppFillEstimates(jaspResults, estimatesMean, models, options, type)
+  }
+
+  # PET-PEESE estimates
+  if (type == "petPeese") {
+    if (is.null(estimates[["petPeese"]]) && options[["estimatesPetPeese"]]) {
+      petPeese <- createJaspTable(title = gettext("PET-PEESE Regression Estimates"))
+      petPeese$position  <- 2
+      petPeese$dependOn(c("estimatesPetPeese"))
+      estimates[["petPeese"]] <- petPeese
+      petPeese <- .wwppFillPetPeese(jaspResults, petPeese, models, options)
+    }
+  }
+
+  # heterogeneity estimates
+  if (is.null(estimates[["heterogeneity"]]) && options[["estimatesSigma"]]) {
+    heterogeneity <- createJaspTable(title = gettext("Multiplicative Heterogeneity Estimates"))
+    heterogeneity$position <- 3
+    heterogeneity$dependOn(c("estimatesSigma"))
+    estimates[["heterogeneity"]] <- heterogeneity
+    heterogeneity <- .wwppFillHeterogeneity(jaspResults, heterogeneity, models, options, type)
+  }
+
+  return()
+}
+# create the plots
+.wwppRegressionPlot          <- function(jaspResults, dataset, options, type = "pet") {
+
+  if (!is.null(jaspResults[[paste0(type, "Regression")]])) {
+    return()
+  } else {
+    plotRegression <- createJaspPlot(
+      title  = gettextf("Estimated %1$s Regression", toupper(type)),
+      width  = 500,
+      height = 400)
+    plotRegression$dependOn(c(.wwppDependencies, switch(type, "pet" = "regressionPet", "peese" = "regressionPeese")))
+    plotRegression$position <- switch(type, "pet" = 5, "peese" = 6)
+    jaspResults[[paste0(type, "Regression")]] <- plotRegression
+  }
+
+  if (!.wwppCheckReady(options))
+    return()
+
+
+  # get the fit
+  fit   <- attr(jaspResults[["models"]]$object[[type]], "fit")
+  fitEs <- attr(jaspResults[["models"]]$object[[type]], "y")
+  fitSe <- attr(jaspResults[["models"]]$object[[type]], "se")
+
+  # get the regression line prediction
+  fitSeRange    <- range(jaspGraphs::getPrettyAxisBreaks(c(0, fitSe)))
+  fitSeSequence <- seq(fitSeRange[1], fitSeRange[2], length.out = 101)
+  fitPrediction <- predict(fit, newdata = data.frame(se = fitSeSequence), interval = "confidence")
+
+  if (options[["measures"]] == "correlation")
+    yLabel <- switch(
+      options[["muTransform"]],
+      "cohensD"  = gettext("Cohen's d"),
+      "fishersZ" = gettext("Fisher's z"))
+  else
+    yLabel <- gettext("Effect Size")
+
+  # make the plot happen
+  plot <- ggplot2::ggplot()
+  # mean
+  plot <- plot + ggplot2::geom_polygon(
```

Comment:
You can just use `+` here instead of `plot <- plot + `

Reply:
fixed here and in other places (hope I didn't miss any)